File: orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/model/DeliveryConfig.java
package com.netflix.spinnaker.orca.front50.model;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import lombok.Data;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Data
public class DeliveryConfig {
  private String id;
  private String application;
  private Long lastModified;
  private Long createTs;
  private String lastModifiedBy;
  private List<Map<String,Object>> deliveryArtifacts;
  private List<Map<String,Object>> deliveryEnvironments;

  private Map<String,Object> details = new HashMap<>();

  @JsonAnyGetter
  Map<String,Object> details() {
    return details;
  }

  @JsonAnySetter
  void set(String name, Object value) {
    details.put(name, value);
  }
}

File: keel-deliveryconfig-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/deliveryconfig/DeliveryConfig.kt
package com.netflix.spinnaker.keel.api.deliveryconfig

data class DeliveryConfig(
  val name: String,
  val application: String,
  val deliveryArtifacts: List<Map<String, Any>> = emptyList(),
  val deliveryEnvironments: List<DeliveryEnvironment> = emptyList()
)
